src/analyzer/trackanalysisscheduler.cpp

```diff
@@ -19,8 +21,11 @@ void deleteTrackAnalysisScheduler(TrackAnalysisScheduler* plainPtr) {
     if (plainPtr) {
         // Trigger stop
         plainPtr->stop();
-        // Release ownership and let Qt delete the queue later
-        plainPtr->deleteLater();
+        if (QCoreApplication::instance()->thread()->loopLevel() > 0) {
```

Comment:
http://doc.qt.io/qt-5/qthread.html#loopLevel  can only be called from the thread itself.

Reply:
Waiting forQCoreApplication::exec() is probably not a good strategy. If the event loop of the object is already down, there can't be pending calls. So we can test for this instead.
Or delete all relevant objects before closing the event loop down.  

It can make also use of some code comments.

Reply:
Oh, I didn't notice this restriction. Initially I was using QThread::currentThread() instead of QCoreApplication::instance()->thread().